controllers/datadogagent/feature/apm/feature.go

```diff
@@ -228,6 +252,47 @@ func (f *apmFeature) ManageDependencies(managers feature.ResourceManagers, compo
 // ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
 // It should do nothing if the feature doesn't need to configure it.
 func (f *apmFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	if f.singleStepInstrumentation != nil && f.singleStepInstrumentation.enabled {
```

Comment:
hmm that would be super confusing UX imo. The flag explicitly says `enabled: false` but then it's not actually disabled? Why should we do that?

Reply:
`singleStepInstrumentation.enabled` is a cluster level configuration, `singleStepInstrumentation.enabledNamespace` is a namespace level configuration. SSI can be disabled in the cluster, but enabled on a specific namespace.

Reply:
Let's not do that please. This would be unique behavior, no other product does that. It's also confusing, as a user you'd think you're disabling the feature but it's not actually the case. Global enable/disable should be global, namespace-level config should be namespace level, etc. and global should override everything, that's how every other config of the agent works.